test/parallel/test-stream2-unpipe-leak.js

```diff
@@ -1,73 +0,0 @@
-// Copyright Joyent, Inc. and other Node contributors.
-//
-// Permission is hereby granted, free of charge, to any person obtaining a
-// copy of this software and associated documentation files (the
-// "Software"), to deal in the Software without restriction, including
-// without limitation the rights to use, copy, modify, merge, publish,
-// distribute, sublicense, and/or sell copies of the Software, and to permit
-// persons to whom the Software is furnished to do so, subject to the
-// following conditions:
-//
-// The above copyright notice and this permission notice shall be included
-// in all copies or substantial portions of the Software.
-//
-// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
-// OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
-// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN
-// NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
-// DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR
-// OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE
-// USE OR OTHER DEALINGS IN THE SOFTWARE.
-
-'use strict';
-require('../common');
-const assert = require('assert');
-const stream = require('stream');
-
-const chunk = Buffer.from('hallo');
-
-class TestWriter extends stream.Writable {
-  _write(buffer, encoding, callback) {
-    callback(null);
-  }
-}
-
-const dest = new TestWriter();
-
-// Set this high so that we'd trigger a nextTick warning
-// and/or RangeError if we do maybeReadMore wrong.
-class TestReader extends stream.Readable {
-  constructor() {
-    super({
-      highWaterMark: 0x10000
-    });
-  }
-
-  _read(size) {
-    this.push(chunk);
-  }
-}
-
-const src = new TestReader();
-
-for (let i = 0; i < 10; i++) {
-  src.pipe(dest);
-  src.unpipe(dest);
-}
```

Comment:
I'm unsure what exactly this test is testing. It seems it expects data to flow in the tick after `unpipe`. However, `unpipe` calls `pause()`, and hence no data should be read.